src/pyeval.jl

```diff
@@ -2,24 +2,36 @@ const Py_single_input = 256  # from Python.h
 const Py_file_input = 257
 const Py_eval_input = 258
 
-const _maindict = PyDict{String,PyObject,false}(PyNULL()) # cache of __main__ module dictionary
-function maindict()
-    if ispynull(_maindict.o)
-        _maindict.o = pyincref(@pycheckn ccall((@pysym :PyModule_GetDict), PyPtr, (PyPtr,), pyimport("__main__")))
+const _namespaces = Dict{Module,PyDict{String,PyObject,true}}()
+
+pynamespace(m::Module) =
+    get!(_namespaces, m) do
+        if m === Main
+            return PyDict{String,PyObject,true}(pyincref(@pycheckn ccall((@pysym :PyModule_GetDict), PyPtr, (PyPtr,), pyimport("__main__"))))
+        else
+            ns = PyDict{String,PyObject}()
+            # In Python 2, it looks like `__builtin__` (w/o 's') must
+            # exist at module namespace.  See also:
+            # http://mail.python.org/pipermail/python-dev/2001-April/014068.html
+            # https://github.com/ipython/ipython/blob/512d47340c09d184e20811ca46aaa2f862bcbafe/IPython/core/interactiveshell.py#L1295-L1299
+            if pyversion < v"3"
+                PyObject(ns).setdefault("__builtin__", builtin)
+                PyObject(ns).setdefault("__builtins__", builtin)
```

Comment:
We should still have `__builtins__` in Python 3,  think?

Reply:
See https://docs.python.org/3/library/builtins.html

Reply:
Good point, I changed it.